Integrations/python/test/testKafkaTools.py

```diff
@@ -58,14 +68,99 @@ def testSimple(self):
-        self.assertEquals("KafkaTimestamp", cols[2].getName())
-        self.assertEquals(dh.datetime.clazz(), cols[2].getDataType())
+        self._assertCommonCols(cols)
         self.assertEquals("Price", cols[3].getName())
         self.assertEquals(dh.double.clazz(), cols[3].getDataType())
         
         del t
 
+    def testJson(self):
+        """
+        Check a JSON Kafka subscription creates the right table.
+        """
+
+        t = kt.consumeToTable(
+            {'bootstrap.servers' : 'redpanda:29092'},
+            'orders',
+            key = kt.IGNORE,
+            value = kt.json(
+                [ ('Symbol', dh.string),
+                  ('Side', dh.string),
+                  ('Price', dh.double),
+                  ('Qty', dh.int_) ],
+                mapping = {
+                    'jsymbol' : 'Symbol',
+                    'jside' : 'Side',
+                    'jprice' : 'Price',
+                    'jqty' : 'Qty'}
+            ),
+            table_type = 'append'
+        )
+        self.assertIsNotNone(t)
+
+        cols = t.getDefinition().getColumns()
+        self.assertEquals(7, len(cols))
+        self._assertCommonCols(cols)
+
+        self.assertEquals("Symbol", cols[3].getName())
+        self.assertEquals(dh.string.clazz(), cols[3].getDataType())
+        self.assertEquals("Side", cols[4].getName())
+        self.assertEquals(dh.string.clazz(), cols[4].getDataType())
+        self.assertEquals("Price", cols[5].getName())
+        self.assertEquals(dh.double.clazz(), cols[5].getDataType())
+        self.assertEquals("Qty", cols[6].getName())
+        self.assertEquals(dh.int_.clazz(), cols[6].getDataType())
+        
+        del t
+
+    def testAvro(self):
+        """
+        Check an Avro Kafka subscription creates the right table.
+        """
+
+        avro_as_json = \
+        """
+        { "type" : "record",
+          "namespace" : "io.deephaven.examples",
+          "name" : "share_price",
+          "fields" : [
+            { "name" : "Symbol", "type" : "string" },
+            { "name" : "Side",   "type" : "string" },
+            { "name" : "Qty",    "type" : "int"    },
+            { "name" : "Price",  "type" : "double" }
+          ]
+        }
+        """
+
+        sys_str = "curl -X POST -H 'Content-type: application/json; artifactType=AVRO' " + \
+          "-H 'X-Registry-ArtifactId: share_price_record' -H 'X-Registry-Version: 1' " + \
+          "--data-binary '%s' http://registry:8080/api/artifacts" % (avro_as_json)
+
+        r = os.system(sys_str)
+        self.assertEquals(0, r)
+        
+        t = kt.consumeToTable(
+            {'bootstrap.servers' : 'redpanda:29092',
+             'schema.registry.url' : 'http://registry:8080/api/ccompat'},
+            'share_price',
+            key = kt.IGNORE,
+            value = kt.avro('share_price_record', schema_version='1'),
+            table_type='append'
+        )
+
+        self.assertIsNotNone(t)
+
```

Comment:
IIRC the constructor would throw an exception when some of the input parameters are invalid, so you have chosen to use exception to report an error. Thus the constructor should never return None. Is this check unnecessary?

Reply:
It is, it had to do with an earlier version before docker compose was wired. Ty